control/gain.go

```diff
@@ -44,7 +44,10 @@ func (b *gain) reset() error {
 	if len(b.cfg.DependsOn) != 1 {
 		return errors.Errorf("invalid number of inputs for gain block %s expected 1 got %d", b.cfg.Name, len(b.cfg.DependsOn))
 	}
-	b.gain = b.cfg.Attribute.Float64("gain", 1.0)
+	b.gain = b.cfg.Attribute["gain"].(float64)
+	if b.gain == 0 {
```

Comment:
q: why only if it's 0? assuming because it defaults to 0 but we want it to default to 1 but just double checking I understand what was happening before

Reply:
If the gain block exists and is zero - it will multiply the entire signal by zero, and we'll lose all the information.

We are checking if it exists in the `cfg.Attribute.Has()` check in line 41, but we do want to explicitly guard a user against setting a zero gain. Maybe refactor this out to error in the future, but guard railing the attribute map is the way to go for now.